### PR TITLE
Added player setAnimation

### PR DESCRIPTION
**Pull request by DaDamRival**
Added player setAnimation and other related methods.

### CODE REVIEW

**Comment by DaDamRival** [comment]
This is clientside only code, and while it would make clientside and serverside hitboxes misaligned. I don't this this would be a big issue considering playGesture is the same in that regard.

**Comment by thegrb93** [comment]
O, I thought it was shared at a glance. clientside is fine.

File: lua/starfall/libs_sh/players.lua [hunk 1]
```diff
@@ -514,6 +545,214 @@ if CLIENT then
...
+	-- @param progress Optional float (Default 0), the progress of the animation. Ranging from 0-1
+	-- @param rate Optional float (Default 1), the playback rate of the animation
+	-- @param loop Optional bool (Default false), should the animation loop
+	-- @param auto_advance Optional bool (Default true), should the animation handle advancing itself
+	-- @param act Optional number or string name (Default sequence value), the activity the player should use
+	function player_methods:setAnimation(seq, progress, rate, loop, auto_advance, act)
+		local ply = getply(self)
+		checkpermission(instance, ply, "entities.setPlayerRenderProperty")
+		
+		if isstring(seq) then
```
Comment (thegrb93) [inline lua/starfall/libs_sh/players.lua:561]: Can you start making checktypes a single line for each param? I'll fix the ones from prior prs later as well.

**Comment by DaDamRival** [comment]
Decided to not compact some lines like the one requested since it would become way to long. Also fixed some methods that should return but didn't.

**Comment by thegrb93** [comment]
The last thing is I'd like CalcMainActivity to only be added while there exists a player with sf_animation.

**Comment by thegrb93** [comment]
I can move its structure to a SF.EntityTable and do that pretty easily.

**Comment by DaDamRival** [comment]
While I was at it also made flex modification use `entities.setPlayerRenderProperty` instead of `entities.setRenderProperty` when applied to a player.

**Comment by thegrb93** [comment]
That permission doesn't check if the player is the owner of the chip so you also need to check that. Otherwise players can affect any players in any client.

**Comment by thegrb93** [comment]
I made EntityTable only contain entity keys now so you can use next() instead of table.count()

File: lua/starfall/libs_sh/players.lua [hunk 2]
```diff
@@ -480,39 +531,233 @@ if CLIENT then
...
+		anim.max = math.min(max, 1)
+		anim.range = anim.max - anim.min
+	end
+	
+	--- Gets whether a animation is playing
+	-- @client
+	-- @return True or false
+	function player_methods:isPlayingAnimation()
+		local ply = getply(self)
+		return playerAnimGet(ply) and true or false
```
Comment (thegrb93) [inline lua/starfall/libs_sh/players.lua:738]: Can you change this to playerAnimGet(ply)~=nil

File: lua/starfall/libs_sh/players.lua [hunk 3]
```diff
@@ -480,39 +531,233 @@ if CLIENT then
...
+	-- @client
+	-- @param loop Should the animation loop. Bool
+	function player_methods:setAnimationLoop(loop)
+		local ply = getply(self)
+		checkpermission(instance, ply, "entities.setPlayerRenderProperty")
+		
+		local anim = playerAnimGet(ply)
+		if not anim then SF.Throw("No animation is playing.", 2) end
+		
+		anim.loop = loop and true or false
```
Comment (thegrb93) [inline lua/starfall/libs_sh/players.lua:711]: A checktype TYPE_BOOLEAN would be better since users inserting a 0,1 expecting it to work would be confused.

**Pull request merged**